### PR TITLE
possibly fix #101

### LINKED ISSUE #101: 0.6: ERROR: GitError(Code:ENOTFOUND, Class:Tree, the path '0.1.0' does not exist in the given tree)
```julia
julia> using PkgDev

julia> PkgDev.publish()
WARNING: cat{T <: GitObject}(repo::GitRepo, ::Type{T}, object::AbstractString) is deprecated, use cat(repo, object) instead.
Stacktrace:
 [1] depwarn(::String, ::Symbol) at ./deprecated.jl:64
 [2] cat(::Base.LibGit2.GitRepo, ::Type{Base.LibGit2.GitBlob}, ::String) at ./deprecated.jl:51
 [3] (::PkgDev.Entry.##4#9{String,Dict{String,Array{String,1}},String})(::Base.LibGit2.GitRepo) at /Users/solver/.julia/v0.6/PkgDev/src/entry.jl:80
 [4] with(::PkgDev.Entry.##4#9{String,Dict{String,Array{String,1}},String}, ::Base.LibGit2.GitRepo) at ./libgit2/types.jl:576
 [5] publish(::String, ::String) at /Users/solver/.julia/v0.6/PkgDev/src/entry.jl:64
 [6] publish() at /Users/solver/.julia/v0.6/PkgDev/src/PkgDev.jl:70
 [7] eval(::Module, ::Any) at ./boot.jl:235
 [8] eval_user_input(::Any, ::Base.REPL.REPLBackend) at ./REPL.jl:66
 [9] macro expansion at ./REPL.jl:97 [inlined]
 [10] (::Base.REPL.##1#2{Base.REPL.REPLBackend})() at ./event.jl:73
while loading no file, in expression starting on line 0
ERROR: GitError(Code:ENOTFOUND, Class:Tree, the path '0.1.0' does not exist in the given tree)
Stacktrace:
 [1] macro expansion at ./libgit2/error.jl:99 [inlined]
 [2] Base.LibGit2.GitObject(::Base.LibGit2.GitRepo, ::String) at ./libgit2/repository.jl:127
 [3] cat at ./libgit2/libgit2.jl:479 [inlined]
 [4] cat(::Base.LibGit2.GitRepo, ::Type{Base.LibGit2.GitBlob}, ::String) at ./deprecated.jl:53
 [5] (::PkgDev.Entry.##4#9{String,Dict{String,Array{String,1}},String})(::Base.LibGit2.GitRepo) at /Users/solver/.julia/v0.6/PkgDev/src/entry.jl:80
 [6] with(::PkgDev.Entry.##4#9{String,Dict{String,Array{String,1}},String}, ::Base.LibGit2.GitRepo) at ./libgit2/types.jl:576
 [7] publish(::String, ::String) at /Users/solver/.julia/v0.6/PkgDev/src/entry.jl:64
 [8] publish() at /Users/solver/.julia/v0.6/PkgDev/src/PkgDev.jl:70
```

### PR DESCRIPTION
**Pull request by simonbyrne**
I'm fairly sure this will fix #101.

Okay, it's not the prettiest code, but it should make do until we have a better solution to the problem of querying whether or not an object exists (which might be hard to do now in Base due to the feature freeze).

### CODE REVIEW

File: src/entry.jl [hunk 1]
```diff
@@ -77,8 +77,17 @@ function publish(branch::AbstractString, prbranch::AbstractString="")
             m !== nothing && ismatch(Base.VERSION_REGEX, m.captures[2]) || continue
             pkg, ver = m.captures; ver = convert(VersionNumber,ver)
             sha1 = readchomp(joinpath(metapath,path))
-            old = LibGit2.cat(repo, LibGit2.GitBlob, "origin/$branch:$path")
-            old !== nothing && old != sha1 && throw(PkgError("$pkg v$ver SHA1 changed in METADATA – refusing to publish"))
+            if VERSION < v"0.6-"
```
Comment (ararslan) [inline src/entry.jl]: Is it possible to be more specific with the version?

Comment (simonbyrne) [inline src/entry.jl]: probably not, a lot of stuff has changed.

**Comment by tkelman** [comment]
Didn't we have a find or search command for a similar replacement in base? This kind of thing is why we need to return "not found" from API's instead of putting try-catches everywhere.

**Comment by simonbyrne** [comment]
Agreed, but that's a lot more work which might not get done before 0.6.

File: src/entry.jl (hunk 1 above)
Comment (tkelman) [inline src/entry.jl]: when did this specifically change to throwing?

**Comment by tkelman** [comment]
I thought there was an alternative you switched to when something similar came up in a base PR?

**Comment by simonbyrne** [comment]
There was this: https://github.com/JuliaLang/julia/pull/20433, but I changed it to check the index instead of the commit (which I don't think will work in this case).

Another option is to change the behaviour of `LibGit2.cat` back and keep using it: the downside of this is that users would still see the deprecation warning.

**Pull request closed without merging**